Dockerfile

```diff
@@ -1,11 +1,19 @@
-# Base java:8
-FROM java:8
+FROM eclipse-temurin:21-jdk-alpine AS build
+WORKDIR /app
 
-# Add Rest Gateway war to container
-ADD src/target/rest-adapter-service-*.war rest-adapter-service.war
+COPY ./adapter/build.gradle.kts ./adapter/settings.gradle.kts ./adapter/gradlew ./
+COPY ./adapter/gradle/wrapper/ ./gradle/wrapper/
+COPY ./adapter/gradle/libs.versions.toml ./gradle/
+COPY ./adapter/src ./src
 
-# Entry with exec
-ENTRYPOINT exec java $JAVA_OPTS -jar /rest-adapter-service.war
+RUN chmod +x ./gradlew
 
-# Expose default port
-EXPOSE 8080
+RUN ./gradlew clean build -x test -x checkstyleMain -x checkstyleTest -x licenseMain -x licenseTest
+RUN cd ./build/libs && ls -lah
+
+FROM eclipse-temurin:21-jre-alpine
```

Comment:
Have you also manually tested the container, Alpine can sometimes bring unexpected results since it does not use glibc.